File: tempconversion.rb
#!/usr/bin/env ruby
# http://rosettacode.org/wiki/Temperature_conversion

def kelvin_to_c(temp)
  (temp - 273.15).round(2)
end

def kelvin_to_f(temp)
  ((kelvin_to_c(temp) * 1.800) + 32.00).round(2)
end

def kelvin_to_rankine(temp)
  ((kelvin_to_f(temp) - 32) + 491.67).round(2)
end

temperature = 21.0
p "kelvin: #{temperature}"
p "celsius: #{kelvin_to_c(temperature)}"
p "fahrenheit: #{kelvin_to_f(temperature)}"
p "rankine: #{kelvin_to_rankine(temperature)}"

File: temp_conversion.swift
// http://rosettacode.org/wiki/Temperature_conversion

func kelvinToC(temp:Double) -> Double {
  return temp - 273.15
} 

func kelvinToF(temp:Double) -> Double {
  return (kelvinToC(temp) * 1.800) + 32.00
}

func kelvinToRankine(temp:Double) -> Double {
  return (kelvinToF(temp) - 32) + 491.67
}

let temperature = 21.0
println("\(temperature) Kelvin")
println(kelvinToC(temperature))
println(kelvinToF(temperature))
println(kelvinToRankine(temperature))
